File: python/blog/ds/stackFixedArray.py
class StackFixedArray:
    top = 0
    arr = [-1 for i in range(4)]

    def toString(self):
        result = []

        for i in range(self.top):
            result.append(str(self.arr[i]))

        return ','.join(result)

    def push(self, data):
        if self.top >= len(self.arr):
            raise IndexError

        self.arr[self.top] = data
        self.top += 1

    def pop(self):
        if self.empty():
            return None
        
        self.top -= 1
        return self.arr[self.top]

    def empty(self):
        return self.top == 0

    def peek(self):
        if self.empty():
            return None
        return self.arr[self.top - 1]

    def search(self, data):
        for i in range(self.top):
            if self.arr[i] == data:
                return i
        return -1

File: java/src/blog/ds/StackFixedArray.java
package blog.ds;

import java.util.StringJoiner;

public class StackFixedArray {
    int top = 0;

    int[] arr = new int[4];

    public String toString() {
        StringJoiner joiner = new StringJoiner(",");
        for(int i = 0; i < top; i++)
            joiner.add(Integer.toString(arr[i]));

        return joiner.toString();
    }

    public void push(int data) {
        if (this.top >= this.arr.length)
            throw new ArrayIndexOutOfBoundsException();

        this.arr[top++] = data;
    }

    public Integer pop() {
        if (this.empty())
            return null;
        return this.arr[--this.top];
    }

    public boolean empty() {
        return this.top == 0;
    }

    public Integer peek() {
        if (this.empty())
            return null;
        return this.arr[this.top - 1];
    }

    public int search(int data) {
        for (int i = 0; i < this.top; i++) {
            if (data == this.arr[i])
                return i;
        }
        return -1;
    }
}
